tests/cli_commands/test_publish.py

```diff
@@ -89,15 +91,39 @@ def setUp(self) -> None:
     def tearDown(self) -> None:
         shutil.rmtree(self.build_temp_dir)
 
+    @staticmethod
+    def mock_clone_from(url: PathLike, to_path: PathLike, **kwargs: Any):
+        def noop():
+            pass
+
+        repo_mock = MagicMock()
+        config_writer_mock = MagicMock()
+        set_value_mock = MagicMock()
+        set_value_mock.configure_mock(**{"release": noop})
+        config_writer_mock.configure_mock(
+            **{"set_value": lambda x, y, z: set_value_mock}
+        )
+        repo_mock.configure_mock(**{"config_writer": config_writer_mock})
+        return repo_mock
+
     @patch("pathlib.Path.cwd", lambda: goldens_dir_path)
     def test_generate_correct_project(self):
         runner = CliRunner()
-
+        git_mock = MagicMock()  # no override
+        repo_mock = MagicMock()
+        repo_mock.configure_mock(**{"clone_from": self.mock_clone_from})
         with patch(
             "data_pipelines_cli.cli_commands.publish.BUILD_DIR", self.build_temp_dir
-        ), patch("data_pipelines_cli.config_generation.BUILD_DIR", self.build_temp_dir):
-            result = runner.invoke(_cli, ["publish"])
-            self.assertEqual(0, result.exit_code, msg=result.exception)
+        ), patch(
+            "data_pipelines_cli.config_generation.BUILD_DIR", self.build_temp_dir
+        ), patch(
+            "data_pipelines_cli.cli_commands.publish.Git", git_mock
+        ), patch(
+            "data_pipelines_cli.cli_commands.publish.Repo", repo_mock
+        ):
+            result = runner.invoke(_cli, ["publish", "--key_path", "SOME_KEY.txt"])
+            result = runner.invoke(_cli, ["publish", "--key_path", "SOME_KEY.txt"])
```

Comment:
Why doubled?

Reply:
quickest way to increase test coverage :)

Reply:
Ok, I get it. But then I would remove `result = ` from the first line :)